README.md

```diff
@@ -1,97 +1,224 @@
 ```
 
-The `ArchipelagoSession` object is your entrypoint for all operations with the server. The session object exposes various helpers which provide functionality for various objectives you might be trying to achieve.
+The freshly created `ArchipelagoSession` object is the entrypoint for all incoming and outgoing interactions with the server. Keep it in scope for at least the lifetime of the connection. If the room changes ports, or the user needs to connect to a different room, then a new session needs to be created at the new host and port.
 
-### LocationCheckHelper
+## Connect to Room
 
-The location check helper may be used to complete or scout location checks. The API also provides utility functions such as retrieving the name of a location from numerical id.
+Connect to a server at a specific room slot using the following method:
 
 ```csharp
-var session = ArchipelagoSessionFactory.CreateSession("localhost", 38281);
+LoginResult TryConnectAndLogin(
+        string game, // Name of the game implemented by this client, SHOULD match what is used in the world implementation
+        string name, // Name of the slot to connect as (a.k.a player name)
+        ItemsHandlingFlags itemsHandlingFlags, /* One of the following (see AP docs for details):
+                NoItems
+                RemoteItems
+                IncludeOwnItems
+                IncludeStartingInventory
+                AllItems
+            */
+        Version version = null, // Minimum Archipelago world specification version which this client can successfuly interface with
+        string[] tags = null, /* One of the following (see AP docs for details)
+                "AP"
+                "IgnoreGame"
+                "DeathLink"
+                "Tracker"
+                "TextOnly"
+            */
+        string uuid = null, // Unique identifier for this player/client, if null randomly generated
+        string password = null // Password that was set when the room was created
+    );
+```
 
-session.TryConnectAndLogin("Risk of Rain 2", "Ijwu", new Version(2,1,0));
-session.Locations.CompleteLocationChecks(42);
+For example,
+
+```csharp
+LoginResult result = session.TryConnectAndLogin("Risk of Rain 2", "Ijwu", new Version(2,1,0));
+```
+
+Would attempt to connect to a password-less room at the slot `Ijwu`, and report the game `Risk of Rain 2` with a minimum apworld version of `v2.1.0`.
+
+Once connected, you have access to a suite of helper objects which provide an interface for sending/receiving information with the server.
+
+## Asynchronous Connection and Error Handling
+
+Note that `TryConnectAndLogin` issues an outgoing websocket connection attempt which can block the current thread for up to multiple seconds. If the client is implemented via code-injection (e.g. harmony), the best practice is to handle large network operations in a separate thread to prevent the game from hitching/freezing. For example:

```

Comment:
While this is not wrong, there are async methods available, they just require .net 4.0 or higher

`ConnectAsync()`
`LoginAsync(....)`